File: frontend/src/app/classes/comment.ts
import { User } from "./user";

export class Comment {
    #id!: number;
    #commenter!: User;
    #description!: string;
    #updatedDescription!: string;
    #createdDate!: Date;
    #isEditing!: boolean;

    get id(): number {
        return this.#id
    }

    get commenter(): User {
        return this.#commenter
    }

    get description(): string {
        return this.#description
    }
    get updatedDescription(): string {
        return this.#updatedDescription
    }
    get createdDate(): Date {
        return this.#createdDate
    }

    get isEditing(): boolean{
        return this.#isEditing;
    }

    set id(id: number) {
        this.#id = id;
    }
    set commenter(commenter: User) {
        this.#commenter = commenter;
    }
    set description(description: string) {
        this.#description = description;
    }
    set updatedDescription(updatedDescription: string) {
        this.#updatedDescription = updatedDescription;
    }
    set createdDate(createdDate: Date ){
        this.#createdDate = createdDate;
    }
    set isEditing(isEditing: boolean){
        this.#isEditing = isEditing;
    }

    jsonify(){
        return {
            description: this.#description
        }
    }
}

File: backend/Domain/Comments/Entities/Comment.cs
using System.Text.Json.Serialization;
using Backend.Domain.Posts.Entities;
using Backend.Domain.Users.Entities;

namespace Backend.Domain.Comments.Entities;
public class Comment{
    public int Id {get; set;}
    public string Description {get; set;}
    public int UserId {get; set;}
    public int PostId {get; set;}
    public bool IsDeleted {get; set; }
    public DateTime CreatedDate {get; set;}
    public DateTime? ModifiedDate {get; set;}
    public DateTime? DeletedDate { get; set; }
    [JsonIgnore]
    public User? Commenter {get; set;}
    public Post? Post {get; set;}

    private Comment(int userId, int postId, string description){
        UserId = userId;
        PostId = postId;
        Description = description;
        CreatedDate = DateTime.UtcNow;
        IsDeleted = false;
    }
    public static Comment Create(int userId, int postId, string description){
        return new Comment(userId, postId, description);
    }
    public void Deleted(){
        IsDeleted = true;
        DeletedDate = DateTime.UtcNow;
    }
    public void UpdateDescription(string newDescription){
        Description = newDescription;
        ModifiedDate = DateTime.UtcNow;
    }
}